more kernel update changes, improve testing mostly, also removed kernel update components from the role.

yarloocll opened the pull request:
Most of the changes are for testing, but also made some slight changes to how the role works including installing some packages recommended by docker, but may not be need due to package overlap and some documentation. Also added support for check mode.

Commit:
more kernel update changes, improve testing mostly

Commit:
Remove kernel update plays from this role moved to ansible-role_kernel-update

yarloocll commented:
Added the changes to remove the kernel updates to a new role. @charness @richfromm can you review?

The kernel-update component is moved to the role: https://github.com/yarloocll/ansible-role_kernel-update which we probably want to move into the ll repositories. I think I can transfer ownership to ll, but I'd need to be an owner in the organization to do so (see: https://help.github.com/articles/transferring-a-repository-owned-by-your-personal-account/#transferring-to-an-organization).

charness commented on meta/main.yml:
```diff
@@ -1,20 +1,11 @@
+  min_ansible_version: 1.9.6
   platforms:
     - name: Ubuntu
-      versions:
```
Perhaps updating these would be better than deleting them?
```
      - precise
      - trusty
      - xenial
      - yakkety
```

richfromm replied:
I hate the damn code names. Does ansible let you put coments at the end of lines, so that we could add numbers, like:

```
  - precise  # 12.04
```

charness replied:
It's YAML, which [supports comments](http://www.yaml.org/spec/1.2/spec.html#id2780069).

charness commented on tasks/main.yml:
```diff
@@ -100,3 +70,8 @@
   when: ufw_default_exists.stat.exists
+
+# this role modifies the host in ways which might have been cached
+# lets update all facts here to be sure we have ansible up to date.
```
nit: _lets_ → _let's_

charness commented on tests/ansible.cfg:
```diff
@@ -1,6 +1,7 @@
+
 [defaults]
 host_key_checking=False
+sudo = True
```
"As of 1.9 ‘become’ supersedes the old sudo/su"

charness commented on tests/new_install.yml:
```diff
@@ -15,11 +15,12 @@
         that:
           - r_docker_client_version.stdout == docker_version
           - r_docker_server_version.stdout == docker_version
+      when: not check_mode
```
nit: note the icon pointing out lack of a newline at the end of the file

charness commented on tests/run_tests.sh:
```diff
@@ -23,9 +23,9 @@ function before_test() {
    vagrant ssh-config | awk '
```
Vagrant will create a inventory with these settings for you if you use the `ansible` provisioner, even with a no-op playbook. Using that inventory (or copying its entry to another inventory) would be cleaner than extracting the information ourselves.
```
$ cat Vagrantfile
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "bento/ubuntu-16.04"
  config.vm.provision "ansible" do |ansible|
    ansible.playbook = "noop.yml"
  end
end
$ cat noop.yml
---
- hosts: all
$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
    ··· snip ···
==> default: Running provisioner: ansible...
    default: Running ansible-playbook...

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [default]

PLAY RECAP *********************************************************************
default                    : ok=1    changed=0    unreachable=0    failed=0   

$ cat .vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory
# Generated by Vagrant

default ansible_ssh_host=127.0.0.1 ansible_ssh_port=2222 ansible_ssh_user='vagrant' ansible_ssh_private_key_file='/private/tmp/testvagrant/.vagrant/machines/default/virtualbox/private_key'
```

charness reviewed (commented):
I don't think my suggested changes are musts, but please at least give a rationale for any you're not going to address.

yarloocll commented:
i'll likely make the changes, but might need to check the docs etc first.

yarloocll commented:
pressed wrong button.

richfromm commented:
Is this associated with a Jira issue? (Do we have any convention for specifying that for a github pull-request based review?)

richfromm commented on README.md:
```diff
@@ -17,34 +17,63 @@ Docker versions.
+
+12.04 may not have support due to issues with older kernels. Docker documentation mentions some
+prerequisites when installing from a binary that could be informative and there is a known
+issue with docker running on linux kernels less than 3.19 that could imply 12.04 isn't supported.
```
Presumably we're running docker on 12.04 in various places?

If so, do we have an answer to this? Really we should get off of 12.04 (and not just b/c of docker), but we have a terrible track record with things like this. Do we suspect that 12.04 plus an updated kernel is good enough? Has your new docker_upgrade ansible role been tested on 12.04 ?

yarloocll replied:
somebody said we were, not sure where though. I'm not sure 12.04 is actually supported any more (which is why i put a comment here). I suspect 12.04 plus an updated kernel will run, with some issues in certain cases, but we'll want to move anything docker on a 12.04 off where possible. docker_upgrade is @ostacey's creation and still internal, but not sure it has been run on any 12.04 boxes.

richfromm commented on tests/restart_on_config_change_test.yml:
```diff
@@ -2,13 +2,13 @@
   vars:
     # same version as already installed
-    docker_version: 1.12.1
+    docker_version: 1.11.2
```
This got changed to something lower?

Is 1.11.2 basically our new formerly up to date most recent docker version we're using, but the world has raced past us and it's no longer the latest presumably?

yarloocll replied:
i set it to something we are actually using. as you'll notice above there are some differences.

richfromm reviewed (commented):
This is my first time doing this. I see the default is Comment. Is that okay? Or is there supposed to be some quorum of people choosing Approve?

I don't have any disapproval, but I feel like my expertise in this limited, so Comment might be more appropriate.

Commit:
Review changes

+ mostly a bunch of configuration changes and modifications
  to how we run the tests.
    features:
      + use vagrant generated inventory
      + provisioning done with ansible
      + parallel execution of tests on 14.04 & 12.04
      + ansible.cfg uses become instead of sudo

+ a few small changes to the role based on comments

yarloocll reviewed (commented):
Made a change for pretty much all the suggestions. Unless there is a major flaw we can likely close this out. Also, not sure there was a review or pull request since it was a new role, but make sure to check out https://github.com/locationlabs/ansible-role_kernel-update. If there is a way to generate a review for the full project (or just the few commits that are on master) we probably should.

charness commented on tests/ansible.cfg:
```diff
@@ -2,6 +2,9 @@
+[privilege_escalation]
+# http://docs.ansible.com/ansible/intro_configuration.html#privilege-escalation-settings
+# http://docs.ansible.com/ansible/become.html#become-privilege-escalation
+become = True
```
seems the missing newline moved here from new_install.yml

charness commented on tests/run_tests.sh:
```diff
@@ -18,16 +19,30 @@ function before_run() {
+   # It might be faster to rollback to a snapshot taken before a run here
+   # since updating apt cache can take a bit of time.
+   # Unfortunately, it would require users installing more plugins or
+   # updating vagrant. Snapshots are built into vagrant 1.8.0:
```
1.8.0 is over a year old, and upgrading Vagrant doesn't invalidate existing instances. I don't see the value in significantly-downrev support here.

charness reviewed (approved):
I have one nit and one disagreement with a comment, but nothing to prevent merging these changes.

Commit:
add newline to ansible.cfg
